engineio/async_gevent.py

```diff
@@ -52,12 +56,71 @@ def wait(self):
         return self._sock.receive()
 
 
+class uWSGIWebSocket(object):  # pragma: no cover
+    """
+    This wrapper class provides a uWSGI WebSocket interface that is
+    compatible with eventlet's implementation.
+    """
+    def __init__(self, app):
+        self.app = app
+
+    def __call__(self, environ, start_response):
+        if _websocket_available != "uwsgi":
+            raise RuntimeError('You need to use the uWSGI server.')
+        self.environ = environ
+        uwsgi.websocket_handshake()
+        return self.app(self) or []
+
+    def close(self):
+        uwsgi.close()
+
+    def send(self, message):
+        uwsgi.websocket_send(message)
+
+    def wait(self):
+        msg = uwsgi.websocket_recv()
+        if not msg:
+            return None
+        return msg.decode()
+
+    def send_and_wait(self, poll_func):
+        """Calls poll_func(block=False) regularly and sends the messages
+        it returns. Apart from that, it observes the websocket for incoming
+        messages. If one arrives, it returns that. After the returned message
+        was processed, this function can be called again to continue its work.
+        A return of None means that the connection was closed. Afterwards."""
+        while True:
```

Comment:
I'm going to test this, but from the looks of it, this is going to consume CPU, whether there is data traffic or not.  I prefer the approach I'm using, which lets the I/O functions block in another thread until there is something to do.